### PR TITLE
Don't uninstall local broker if in use

### LINKED ISSUE #29: `subctl uninstall` deleted the broker even if it's used
**What happened**:
Deployed 2 clusters on AWS (but this will probably reproduce on KIND), mkolesni-0.12-testday1 had the broker + submariner and mkolesni-0.12-testday2 joined.
When running `subctl uninstall` on mkolesni-0.12-testday1 saw the following:

```bash
 ✓ Deleting the Submariner namespace "submariner-operator" on cluster "mkolesni-0.12-testday1" 
 ✓ Deleting the broker namespace "submariner-k8s-broker" on cluster "mkolesni-0.12-testday1" 
 ✓ Deleting the Submariner custom resource definitions on cluster "mkolesni-0.12-testday1" 
 ✓ Deleted the "brokers.submariner.io" custom resource definition
```

Then when trying to re-join cluster1 I got:
```bash
 subctl join --kubeconfig /path/to/mkolesni-0.12-testday1/auth/kubeconfig broker-info.subm 
* broker-info.subm says broker is at: https://api.mkolesni-0.12-testday1.devcluster.openshift.com:6443
* There are 1 labeled nodes in the cluster:
  - ip-10-0-53-110.ec2.internal
⠈⠑ Discovering network details         Network plugin:  OpenShiftSDN
        Service CIDRs:   [172.30.0.0/16]
        Cluster CIDRs:   [10.128.0.0/14]
 ✓ Discovering network details
Error retrieving broker admin config: Unauthorized

subctl version: v0.12.0-rc1
```

**What you expected to happen**:
It this case `mkolesni-0.12-testday2` was still joined to the broker and thus the broker should've been kept on `mkolesni-0.12-testday1`

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Diagnose information (use `subctl diagnose all`):
- Gather information (use `subctl gather`)
- Cloud provider or hardware configuration:
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by tpantelis**
...by remote clusters. The determination is made by checking if there are any `Endpoint` resources in the broker namespace that don't belong to the local cluster. If it's determined the broker is still in use, leave the broker namespace and operator components.

Fixes https://github.com/submariner-io/subctl/issues/29

### CODE REVIEW

File: pkg/uninstall/all.go [hunk 1]
```diff
@@ -245,3 +242,64 @@ func ensureDeleted(resourceInterface resource.Interface, name string) error {
 		return false, err
 	})
 }
+
+func maybeDeleteBrokerNamespace(clients client.Producer, namespace, clusterName string, status reporter.Interface) (bool, error) {
```
Comment (mkolesnik) [inline pkg/uninstall/all.go]:
```suggestion
func deleteBrokerIfUnused(clients client.Producer, namespace, clusterName string, status reporter.Interface) (bool, error) {
```

File: pkg/uninstall/all.go [hunk 2]
```diff
@@ -245,3 +242,64 @@ func ensureDeleted(resourceInterface resource.Interface, name string) error {
...
+	if namespace == "" {
+		return true, nil
+	}
+
+	_, err := clients.ForKubernetes().CoreV1().Namespaces().Get(context.TODO(), namespace, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return true, nil
+	}
+
+	if err != nil {
```
Comment (mkolesnik) [inline pkg/uninstall/all.go]:
```suggestion
	if err != nil {
		if apierrors.IsNotFound(err) {
			return true, nil
		}

```

File: pkg/uninstall/all.go [hunk 3]
```diff
@@ -245,3 +242,64 @@ func ensureDeleted(resourceInterface resource.Interface, name string) error {
...
+	var remoteClusterNames []string
+
+	for i := range endpoints.Items {
+		if endpoints.Items[i].Spec.ClusterID != clusterName {
+			remoteClusterNames = append(remoteClusterNames, endpoints.Items[i].Name)
+		}
+	}
+
+	if len(remoteClusterNames) > 0 {
+		status.Warning("Broker namespace %q appears to be in use by other clusters - keeping the broker components. Found Endpoints %v",
```
Comment (mkolesnik) [inline pkg/uninstall/all.go]:
```suggestion
		status.Warning("Broker namespace %q appears to be in use by other clusters (%v) - keeping the broker components.",
```

File: pkg/uninstall/all.go [hunk 4]
```diff
@@ -245,3 +242,64 @@ func ensureDeleted(resourceInterface resource.Interface, name string) error {
...
+	endpoints, err := clients.ForSubmariner().SubmarinerV1().Endpoints(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return false, status.Error(err, "error retrieving Endpoints")
+	}
+
+	var remoteClusterNames []string
+
+	for i := range endpoints.Items {
+		if endpoints.Items[i].Spec.ClusterID != clusterName {
+			remoteClusterNames = append(remoteClusterNames, endpoints.Items[i].Name)
```
Comment (mkolesnik) [inline pkg/uninstall/all.go]:
```suggestion
		remoteClusterName := endpoints.Items[i].Spec.ClusterID
		if remoteClusterName != clusterName {
			remoteClusterNames = append(remoteClusterNames, remoteClusterName)
```

**Review by mkolesnik — APPROVED** [review]
Some suggestions but overall looks good

**Review by skitt — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Pull request merged**